core/iwasm/fast-jit/fe/jit_emit_numberic.c

```diff
@@ -353,53 +353,66 @@ compile_int_div_no_check(JitCompContext *cc, IntArithmetic arith_op,
         case INT_DIV_S:
         case INT_DIV_U:
             if (is_i32) {
+                /* workaround for vr->distances == NULL*/
+                GEN_INSN(MOV, eax_hreg, eax_hreg);
+                GEN_INSN(MOV, edx_hreg, edx_hreg);
+
                 GEN_INSN(MOV, eax_hreg, left);
                 if (arith_op == INT_DIV_S)
                     GEN_INSN(DIV_S, eax_hreg, eax_hreg, right);
                 else
                     GEN_INSN(DIV_U, eax_hreg, eax_hreg, right);
-                /* Just to indicate that edx is used,
-                   register allocator cannot spill it out */
+
+                GEN_INSN(MOV, eax_hreg, eax_hreg);
                 GEN_INSN(MOV, edx_hreg, edx_hreg);
                 res = eax_hreg;
             }
             else {
+                /* workaround for vr->distances == NULL*/
+                GEN_INSN(MOV, rax_hreg, rax_hreg);
+                GEN_INSN(MOV, rdx_hreg, rdx_hreg);
```

Comment:
Should be `GEN_INSN(MOV, eax_hreg, eax_hreg)` and GEN_INSN(MOV, edx_hreg, edx_hreg), here it uses i32 register eax/edx implicitly.